### ISSUE TITLE
Crippling FPS Drops (Server-Sided)

### ISSUE THREAD

**Issue by Dragonzrkool** [Mithion/ArsMagica2#1538]
Hello, wasn't sure where to have this problem diagnosed and discussed so I thought best to report it here on GitHub. This isn't necessarily a bug (Don't think at least), or crash as it is a very bizarre problem.

To start off with: We just attempted to play with AM2, latest build today, on our small private server. I am in charge of making sure mods are compatible and functional for the most part - our current build works near flawlessly in singleplayer. I run at 50-60 FPS, dropping to 30 in the most overloaded parts of the world. I do use Optifine and FC for improved performance since I run about 90 mods.

To give some specs there:
I have an AMD FX-8320, Eight-Core 3.5GHz CPU (Though it runs at 4.0)
I have 16 GB DDR3 1337, total RAM. I have given Minecraft 6GB to utilize as it wishes.
My GPU is an ATI Radeon R9 390 - 8GB
Everything else seems irrelevant to list, but I can supply it if need be.

We loaded in AM2 on our server, turning on retrogen for a reloading of our world. However, any player that plays on the server has an immense drop rate in frames when merely in the world. I managed to only get about 1 to 3 FPS when walking around - in both populated and unpopulated areas, including the underground. Going into a menu or opening the chat window for some reason removed this lag issue - restoring my frames to the typical for the time the menus were opened. As soon as I closed them, they dropped to abysmal rates once again.

From here, we decided to do a number of things. We manually turned off retrogen and ensured it was off. We also disabled the leaf particles and consistent memory update feature - opting for packet data instead. The problem still persisted. We tried a different map, fresh from the start without retrogen and still experienced the same lag issues. We downgraded to .008 from last year and for some reason, still had this lag issue. We had multiple clients connect, including myself and the server host - each player experienced the same exact issue; additionally, we connected at different times as well. As a final measure, we pulled out some server utilities, but the issue still persists. To be clear: This problem **cannot** be replicated in singleplayer, everyone runs at their standard typical FPS. It only applies to the server, and when not in a menu or chat window. Removing AM2 fixes the issue completely, and everything returns to normal.

I was told the server is being run on a dedicated 4 GB RAM, Intel Dual-Quad Core at 3.9GHz. We have never had this problem prior to adding AM2.

We are somewhat unsure what the issue is. I had ran a TPS and Ping test to see if there was any latency or tick issues. Server runs at 20 TPS, with an average tick time of 10-15ms. The only player with any latency was me, and that's expected connecting from a different country. The only thing we can come to some conclusion about is a compatibility issue, but we are unsure. I've attached a screenshot of my mods folder [With AM2 pulled out. The server runs an exact duplicate, but with ForgeEssentials added and FC/Optifine pulled out], as well as the most recent 
logs on the server. Hopefully someone here may make more sense of this problem than us.

![screenshot 22](https://cloud.githubusercontent.com/assets/16041463/16554574/23cfb01a-4213-11e6-8854-db5e29a2521c.png)
![screenshot 23](https://cloud.githubusercontent.com/assets/16041463/16554589/3ab3e8be-4213-11e6-8e7a-2609734ebaf4.png)
![screenshot 24](https://cloud.githubusercontent.com/assets/16041463/16554594/40245586-4213-11e6-9899-9636698872e7.png)

https://drive.google.com/file/d/0B5RN1tBO7O1RWXlwSUcycUhHYlE/view?usp=sharing

Thanks,
Dragonzrkool

**Comment by Thefloofiestwoof** [comment]
@Mithion I'm the server admin of the aforementioned private server, and short of taking out our 90-ish mods and removing them one at a time to see which is causing an issue, we're at a loss. Would really appreciate if you could take a look at this, or if you're aware of some mod conflict we've overlooked to please let us know. Thanks for your time.

**Comment by TheIcyOne** [comment]
Is this still an issue without Optifine and Fastcraft installed?

**Comment by Thefloofiestwoof** [comment]
@TheIcyOne Yeah, those were one of the first things to go in ongoing attempts to fix this bug, so far nothing's changed. This morning I tried removing Project:Red and all of its components, as well as removing other plugins that seem to influence world-gen, no luck on that front either

**Comment by TheIcyOne** [comment]
Would you be able to load the world in SSP, to see if the same issue occurs?

**Comment by Thefloofiestwoof** [comment]
@TheIcyOne In survival single player, there's no lag whatsoever, with all the mods enabled.

**Comment by Dragonzrkool** [comment]
@TheIcyOne To add to this, the result is the same for everyone in survival single player - no FPS drops at all. And the result is likewise the same for anyone on his server: Everyone has crippling FPS unless they're in a menu or the chat menu is open.

**Comment by Thefloofiestwoof** [comment]
**Quick update on this issue:** I've narrowed down the lag to chunk-loading, which seems to be a constant thing in minecraft when you're moving, turning, etc. Again, no issue in single player, but on the server (which has plenty of ram, not the issue) there's unplayable lag when moving/turning too quickly/teleporting etc.

**Comment by TheIcyOne** [comment]
Which Forge version and Java versions are the server running?

**Comment by Thefloofiestwoof** [comment]
The server is running the latest version of java, and using Forge 1.7.10-10.13.4.1614. Additionally, I tested to see if the server just needed more ram than the 5GB its running with, and changing it didn't do anything to alleviate the lag- in fact, it made it worse. Trying with less ram didn't fix anything, though.

**Comment by jjtParadox** [comment]
@Thefloofiestwoof can you use the mod [Opis](https://minecraft.curseforge.com/projects/opis) to profile the client and server to figure out what's taking the most time?

Otherwise, please use [this guide](http://ae-mod.info/Profiling/) and other profiling guides to use Java's profiler to look for more low level information.
